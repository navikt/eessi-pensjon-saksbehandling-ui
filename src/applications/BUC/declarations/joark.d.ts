
export interface JoarkFileVariant {
  variantformat:string;
  filnavn: string;
}

export interface JoarkDoc {
  dokumentInfoId: string;
  tittel: string;
  dokumentvarianter: Array<JoarkFileVariant>;
}
export interface JoarkPoster {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: string;
  tilleggsopplysninger?: string;
  dokumenter: Array<JoarkDoc>
}

export interface JoarkFile {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: Date;
  dokumentInfoId: string;
  tilleggsopplysninger?: string;
  variant: JoarkFileVariant
}

interface Content {
  base64: string;
}

export interface JoarkFileWithContent extends JoarkFile {
  content: Content;
  name: string;
  size: number;
  mimetype: string;
}
