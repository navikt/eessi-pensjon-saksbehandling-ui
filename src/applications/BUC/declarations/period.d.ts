export interface PeriodDate {
  day: string;
  month: string;
  year: string;
}

export interface Country {
  label: string;
  value: string;
}

export interface Period {
  id?: number,
  type: string;
  startDate: PeriodDate;
  endDate?: PeriodDate;
  dateType: string;
  uncertainDate: boolean;
  country: Country;
  comment: string;
  otherType?: string;
  insuranceName?: string;
  insuranceType?: string;
  insuranceId ?: string;
  workActivity ?: string;
  workName ?: string;
  workType ?: string;
  workStreet ?: string;
  workCity ?: string;
  workZipCode ?: string;
  workRegion ?: string;
  childFirstName ?: string;
  childLastName ?: string;
  childBirthDate ?: PeriodDate;
  learnInstitution ?: string;
  payingInstitution ?: string;
  attachments ?: Array[any];
  [key: string]: any;
}

export type Periods = Array<Period>

export type PeriodErrors = {[k: string]: string | undefined}

export interface PayloadPeriod {
  land: string;
  periode: null | {
    lukketPeriode?: {
      fom: string | null;
      tom: string | null;
    }
    openPeriode?:  {
      fom: string | null;
      extra: string;
    }
  };
  vedlegg ?: Array[any];
  trygdeordningnavn ?: string;
  medlemskap ?: string;
  forsikkringEllerRegistreringNr ?: string;
  annenInformasjon ?: string;
  usikkerDatoIndikator ?: string;
  jobbUnderAnsattEllerSelvstendig ?: string;
  navnFirma ?: string;
  typePeriode ?: string;
  adresseFirma ?: {
    postnummer: string | undefined;
    by: string | undefined;
    land: string;
    gate: string | undefined;
    region: string | undefined;
  };
  navnPaaInstitusjon ?: string;
  informasjonBarn ?:  {
    etternavn: string | null | undefined;
    fornavn: string | null | undefined;
    foedseldato: string | null | undefined;
    land: string;
  }
}

export interface Payload {
  ansattSelvstendigPerioder ?: Array<PayloadPeriod>;
  boPerioder ?: Array<PayloadPeriod>;
  barnepassPerioder ?: Array<PayloadPeriod>;
  frivilligPerioder ?: Array<PayloadPeriod>;
  forsvartjenestePerioder ?: Array<PayloadPeriod>;
  foedselspermisjonPerioder ?: Array<PayloadPeriod>;
  opplaeringPerioder ?: Array<PayloadPeriod>;
  arbeidsledigPerioder ?: Array<PayloadPeriod>;
  sykePerioder ?: Array<PayloadPeriod>;
  andrePerioder ?: Array<PayloadPeriod>;
}

export interface P4000Info {
  person: any,
  bank: any,
  stayAbroad: Array<Period>
}

export interface P4000PayloadInfo {
  personInfo: any,
  bankInfo: any,
  periodeInfo: Array<PayloadPeriod>
}
