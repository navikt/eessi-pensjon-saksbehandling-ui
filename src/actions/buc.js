
const BUCLIST = {

  ONGOING: [
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ']
    },
    {
      type: 'P_BUC_02',
      name: 'UførePensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['SE', 'DK', 'CZ'],
      merknader: ['foo', 'bar', 'baz']
    },
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ'],
      comments: ['foo', 'bar', 'baz']
    },
    {
      type: 'P_BUC_01',
      name: 'AldersPensjon',
      dateCreated: 'dd.mm.åå',
      countries: ['ZW', 'KR', 'SE', 'DK', 'CZ'],
      merknader: ['foo', 'bar', 'baz'],
      comments: ['foo', 'bar', 'baz']
    }
  ],
  OTHER: [
    {
      type: 'P_BUC_01',
      name: 'FIFOFA',
      dateCreated: 'dd.mm.åå',
      countries: ['RU', 'GB', 'CH']
    },
    {
      type: 'P_BUC_02',
      name: 'GRÅTASS',
      dateCreated: 'dd.mm.åå',
      countries: ['DE', 'TR', 'CA']
    }
  ]
}

const SEDS = [
  {
    name: 'P2000',
    status: 'sent',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P3000SE',
    status: 'draft',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P4000',
    status: 'received',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  },
  {
    name: 'P5000',
    status: 'foo',
    date: 'dd.mm.åå',
    country: 'Sverige',
    institution: 'Försäkringskassan'
  }
]

export const fetchBucList = async () => {
  return BUCLIST
}

export const fetchSedListForBuc = async (buc) => {
  return SEDS
}
