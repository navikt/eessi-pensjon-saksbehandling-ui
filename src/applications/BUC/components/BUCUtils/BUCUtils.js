import { CountryData } from 'eessi-pensjon-ui'

export const getBucTypeLabel = ({ type, locale, t }) => {
  if (!type.match('P3000_')) {
    return t('buc:buc-' + type)
  }
  const country = type.match(/_(.*)$/)[1]
  const countryLabel = CountryData.findByValue(locale, country)
  return t('buc:buc-P3000_XX', { country: countryLabel.label })
}

// the higher the indexOf, the higher it goes in the sorted list
const sedTypes = ['X', 'H', 'P']

export const sedSorter = (a, b) => {
  if (b.lastUpdate - a.lastUpdate > 0) return 1
  if (b.lastUpdate - a.lastUpdate < 0) return -1
  const mainCompare = parseInt(a.type.replace(/[^\d]/g, ''), 10) - parseInt(b.replace.type(/[^\d]/g, ''), 10)
  const sedTypeA = a.type.charAt(0)
  const sedTypeB = b.type.charAt(0)
  if (sedTypes.indexOf(sedTypeB) - sedTypes.indexOf(sedTypeA) > 0) return 1
  if (sedTypes.indexOf(sedTypeB) - sedTypes.indexOf(sedTypeA) < 0) return -1
  return mainCompare
}
