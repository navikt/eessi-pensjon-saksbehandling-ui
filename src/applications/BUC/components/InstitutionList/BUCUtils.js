import { CountryData } from 'eessi-pensjon-ui'

export const getBucTypeLabel = ({ type, locale, t }) => {
  if (!type.match('P3000_')) {
    return t('buc:buc-' + type)
  }
  const country = type.match(/_(.*)$/)[1]
  console.log(type, country)
  const countryLabel = CountryData.findByValue(locale, country)
  return t('buc:buc-P3000_XX', { country: countryLabel.label })
}
