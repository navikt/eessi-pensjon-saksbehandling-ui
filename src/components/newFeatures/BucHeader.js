import React from 'react'

import * as Typografi from 'nav-frontend-typografi'
import Tekstomrade from 'nav-frontend-tekstomrade'
import { LenkepanelBase } from 'nav-frontend-lenkepanel';


import FlagList from './FlagList'

import { ReactComponent as ProblemCircle } from '../../resources/images/report-problem-circle.svg'
import { ReactComponent as BubbleChat } from '../../resources/images/bubble-chat-2.svg'

import './BucHeader.css'

function preventDefault(onClick){
    return function onClickHandler(e){
        e.stopPropagation()
        e.preventDefault()
        onClick(e)
    }
}

function BucHeader(props){
    const {t} = props
    return (
    <div className='p-0 w-100 d-flex justify-content-between'>
        <div className='col-3'>
            <Typografi.Ingress data-qa='BucHeader-type-name' className='font-weight-bold'>{props.type + ' - ' + props.name}</Typografi.Ingress>
            <Typografi.Normaltekst data-qa='BucHeader-dateCreated'>{t('created')}: {props.dateCreated}</Typografi.Normaltekst>
        </div>
        <div className='col-3 mr-auto d-flex align-items-center pl-0 pr-0'>
            <FlagList data-qa='BucHeader-FlagList' countries={props.countries} overflowLimit={2} flagPath='../../../../flags/' extention='.png' />
        </div>
        <div className='col-3 d-flex justify-content-end align-items-center pl-0 pr-0'>
            <LenkepanelBase data-qa='BucHeader-LinkButton' onClick={preventDefault(props.behandlingOnClick)} className='knapp knapp--mini pl-3 pr-3 pt-2 pb-2 m-0' href={props.href} border>
                <div>
                    {t('processing')}
                </div>
            </LenkepanelBase>
        </div>
        <div className='col-2 d-flex justify-content-end align-items-center'>
            {props.merknader.length > 0
            ? <ProblemCircle data-qa='BucHeader-ProblemCircle' className='c-ui-bucheader-svg mr-2 ml-2'/>
            : null
            }
            {props.comments.length > 0
            ? <BubbleChat data-qa='BucHeader-BubbleChat' className='c-ui-bucheader-svg mr-2 ml-2'/>
            : null
            }
        </div>
    </div>
    )
}

BucHeader.defaultProps = {
    type: '',
    name: '',
    dateCreated: '',
    href: '', 
    countries: [],
    merknader: [],
    comments: [],
    behandlingOnClick: ()=>{},
    t: arg=>arg
}

export default BucHeader

