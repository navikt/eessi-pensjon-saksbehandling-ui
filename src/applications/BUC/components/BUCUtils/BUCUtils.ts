import * as constants from 'constants/constants'
import { JoarkBrowserItem } from 'declarations/joark.d'
import { AllowedLocaleString, Option, PesysContext, T } from 'declarations/app.d'
import { PersonAvdodPDL, PersonAvdodsPDL, PersonPDL } from 'declarations/person.d'
import CountryData, { Country, CountryFilter } from 'land-verktoy'
import _ from 'lodash'
import { Buc, Sed, ValidBuc } from 'declarations/buc'
import moment from 'moment'

interface getBucTypeLabelProps {
  type: string
  locale: AllowedLocaleString
  t: T
}

// the higher the indexOf, the higher it goes in the sorted list
const sedTypes: Array<string> = ['X', 'H', 'P']

export const bucSorter = (a: Buc, b: Buc): number => {
  return moment((a as ValidBuc).startDate).isSameOrAfter(moment((b as ValidBuc).startDate)) ? -1 : 1
}

export const bucFilter = (buc: Buc): boolean => {
  return !_.isEmpty(buc.error) || (
    buc.type
      ? (
          buc.type.startsWith('P_BUC') || _.includes(['H_BUC_07', 'R_BUC_01', 'R_BUC_02', 'M_BUC_02', 'M_BUC_03a', 'M_BUC_03b'], buc.type)
        )
      : true
  )
}

export const bucsThatSupportAvdod = (bucType: string | null | undefined): boolean => _.includes(['P_BUC_02', 'P_BUC_05', 'P_BUC_10'], bucType)

export const countrySorter = (locale: string) => {
  return (a: string, b: string): number => {
    const countryInstance: any = CountryData.getCountryInstance(locale)
    const labelA = countryInstance.findByValue(a)
    const labelB = countryInstance.findByValue(b)
    if (CountryFilter.SCANDINAVIA.indexOf(b.toUpperCase()) - CountryFilter.SCANDINAVIA.indexOf(a.toUpperCase()) > 0) return 1
    if (CountryFilter.SCANDINAVIA.indexOf(b.toUpperCase()) - CountryFilter.SCANDINAVIA.indexOf(a.toUpperCase()) < 0) return -1
    return labelA.label.localeCompare(labelB.label)
  }
}

export const getBucTypeLabel = ({ type, locale, t }: getBucTypeLabelProps): string => {
  if (type && !type.match('P3000_')) {
    return t('buc:buc-' + type)
  }
  const re: RegExpMatchArray | null = type ? type.match(/_(.*)$/) : null
  if (!re) {
    return ''
  }
  const countryLabel: Country | undefined = CountryData.getCountryInstance(locale).findByValue(re[1])
  return t('buc:buc-P3000_XX', { country: countryLabel?.label || t('ui:unknownLand') })
}

export const getFnr = (p: PersonPDL | null | undefined): string | undefined => _.find(p?.identer, i => i.gruppe === 'FOLKEREGISTERIDENT')?.ident

export const getRelasjonTilPerson = (p: PersonPDL | null | undefined, needle: string | undefined) => _.find(p?.familierelasjoner, r => r.relatertPersonsIdent === needle)?.relatertPersonsRolle

export const getRelasjonTilAvdod = (pa: PersonAvdodPDL | null | undefined, needle: string | undefined) => _.find(pa?.familierelasjoner, r => r.relatertPersonsIdent === needle)?.minRolleForPerson

export const labelSorter = (a: Option, b: Option) => a.label.localeCompare(b.label)

export const pbuc02filter = (pesysContext: PesysContext, personAvdods: PersonAvdodsPDL | undefined) =>
  (buc: Buc) => {
    if (buc.type === 'P_BUC_02' && pesysContext !== constants.VEDTAKSKONTEKST && (
      // 'NO:NAVAT08' in test environment should be read as a foreign institution
      buc?.creator?.country === 'NO' && (buc?.creator?.institution !== 'NO:NAVAT06' && buc?.creator?.institution !== 'NO:NAVAT08')
    )) {
      return false
    }
    if (buc.type === 'P_BUC_02' && pesysContext === constants.VEDTAKSKONTEKST &&
      personAvdods?.length === 0 && buc?.creator?.country === 'NO') {
      return false
    }
    return true
  }

export const renderAvdodName = (avdod: PersonAvdodPDL | null | undefined, person: PersonPDL | null | undefined, t: Function): string | undefined => {
  if (!avdod || !person) {
    return undefined
  }
  const fnr = getFnr(avdod)
  const personFnr = getFnr(person)
  return avdod?.navn?.fornavn +
    (avdod?.navn?.mellomnavn ? ' ' + avdod?.navn?.mellomnavn : '') +
    (avdod?.navn?.etternavn ? ' ' + avdod?.navn?.etternavn : '') +
    ' - ' + fnr + ' (' + t('buc:relasjon-' + getRelasjonTilAvdod(avdod, personFnr)) + ')'
}

export const sedAttachmentSorter = (a: JoarkBrowserItem, b: JoarkBrowserItem): number => {
  if (b.type === 'joark' && a.type === 'sed') return -1
  if (b.type === 'sed' && a.type === 'joark') return 1
  return b.key.localeCompare(a.key)
}

export const sedFilter = (sed: Sed): boolean => {
  return sed.status !== 'empty'
}

export const sedSorter = (a: Sed, b: Sed): number => {
  if (b.lastUpdate - a.lastUpdate > 0) return 1
  if (b.lastUpdate - a.lastUpdate < 0) return -1
  const mainCompare = parseInt(a.type.replace(/[^\d]/g, ''), 10) - parseInt(b.type.replace(/[^\d]/g, ''), 10)
  const sedTypeA = a.type.charAt(0)
  const sedTypeB = b.type.charAt(0)
  if (sedTypes.indexOf(sedTypeB) - sedTypes.indexOf(sedTypeA) > 0) return 1
  if (sedTypes.indexOf(sedTypeB) - sedTypes.indexOf(sedTypeA) < 0) return -1
  return mainCompare > 0 ? 1 : -1
}

export const valueSorter = (a: Option, b: Option) => a.value.localeCompare(b.value)
