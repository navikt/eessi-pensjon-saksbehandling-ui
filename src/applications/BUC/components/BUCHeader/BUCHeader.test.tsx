import { sedFilter } from 'applications/BUC/components/BUCUtils/BUCUtils'
import { Buc, BucsInfo } from 'declarations/buc'
import { mount, ReactWrapper } from 'enzyme'
import mockBucs from 'mocks/buc/bucs'
import mockBucsInfo from 'mocks/buc/bucsInfo'
import moment from 'moment'
import { stageSelector } from 'setupTests'
import BucHeader, { BUCHeaderProps, BUCHeaderSelector } from './BUCHeader'

const defaultSelector: BUCHeaderSelector = {
  gettingBucDeltakere: false,
  institutionNames: {},
  locale: 'nb',
  rinaUrl: 'http://rinaurl.mock.com',
  size: 'lg'
}

describe('applications/BUC/components/BUCHeader/BUCHeader', () => {
  let wrapper: ReactWrapper
  const buc: Buc = mockBucs()[0]
  buc.deltakere = buc.institusjon
  const initialMockProps: BUCHeaderProps = {
    buc: buc,
    bucInfo: (mockBucsInfo as BucsInfo).bucs['' + buc.caseId],
    newBuc: true
  } as BUCHeaderProps

  beforeEach(() => {
    stageSelector(defaultSelector, {})
    wrapper = mount(<BucHeader {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__' + buc.type + '-' + buc.caseId + '\']')).toBeTruthy()
    expect(wrapper.find('[data-test-id=\'a-buc-c-bucheader__title-id\']').hostNodes().render().text()).toEqual('P_BUC_01 - buc:buc-P_BUC_01')
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__label-id\']')).toBeTruthy()
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__label-date-id\']')).toBeTruthy()
    expect(wrapper.find('[data-test-id=\'a-buc-c-bucheader__label-date-id\']').hostNodes().render().text()).toEqual(
      'ui:created: ' + moment(new Date(buc.startDate as number)).format('DD.MM.YYYY')
    )
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__label-owner-id\']')).toBeTruthy()
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__label-owner-institution-id\']')).toBeTruthy()
    expect(wrapper.find('[data-test-id=\'a-buc-c-bucheader__label-owner-id\']').hostNodes().render().text()).toEqual(
      'buc:form-caseOwner: NO:NAVAT07'
    )
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__label-case-id\']')).toBeTruthy()
    expect(wrapper.find('[data-test-id=\'a-buc-c-bucheader__label-case-id\']').hostNodes().render().text()).toEqual('buc:form-caseNumberInRina: 600891')

    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__label-avdod-id\']')).toBeFalsy()

    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__icon-id\']')).toBeTruthy()
    expect(wrapper.find('[data-test-id=\'a-buc-c-bucheader__icon-id\'] FlagList')).toBeTruthy()

    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__icon-numberofseds-id\']')).toBeTruthy()

    expect(wrapper.find('[data-test-id=\'a-buc-c-bucheader__icon-numberofseds-id\']').hostNodes().render().text()).toEqual('' + buc.seds?.filter(sedFilter).length)
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__icon-tags-id\']')).toBeTruthy()
    expect(wrapper.find('[data-test-id=\'a-buc-c-bucheader__icon-tags-id\'] ProblemCircleIcon')).toBeTruthy()
  })

  it('Render: shows icon if we have tags', () => {
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__icon-tags-id\']')).toBeTruthy()
    wrapper.setProps({
      bucInfo: {
        tags: [],
        comment: undefined
      },
      buc: {
        ...buc,
        seds: []
      }
    })
    expect(wrapper.exists('[data-test-id=\'a-buc-c-bucheader__icon-tags-id\']')).toBeFalsy()
  })

  it('Handling: open Rina link', () => {
    window.open = jest.fn()
    wrapper.find('[data-test-id=\'a-buc-c-bucheader__label-case-gotorina-link-id\']').first().simulate('click')
    expect(window.open).toHaveBeenCalledWith(defaultSelector.rinaUrl! + initialMockProps.buc.caseId, 'rinaWindow')
  })
})
