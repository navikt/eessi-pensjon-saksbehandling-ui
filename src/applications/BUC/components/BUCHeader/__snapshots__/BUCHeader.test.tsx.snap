// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/BUCHeader/BUCHeader Render: match snapshot 1`] = `
@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {

}

@supports not (-ms-high-contrast:none) {

}

@media (min-width:768px) {

}

@media (min-width:768px) {

}

<div
  class="navds-link-panel__title navds-heading navds-heading--medium"
>
  <h1
    class="lenkepanel--heading navds-heading navds-heading--small"
    data-testid="a_buc_c_BUCHeader--title_id"
  >
    P_BUC_01 - buc:buc-P_BUC_01
  </h1>
</div>
`;
